streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorderTest.java

```diff
@@ -40,105 +46,312 @@
 @RunWith(PowerMockRunner.class)
 @PrepareForTest({RocksDBMetrics.class, Sensor.class})
 public class RocksDBMetricsRecorderTest {
-
     private final static String METRICS_SCOPE = "metrics-scope";
-    private final static String STORE_NAME = "store name";
-    private final static String SEGMENT_STORE_NAME_1 = "segment store name 1";
-    private final static String SEGMENT_STORE_NAME_2 = "segment  name 2";
+    private final static String STORE_NAME = "store-name";
+    private final static String SEGMENT_STORE_NAME_1 = "segment-store-name-1";
+    private final static String SEGMENT_STORE_NAME_2 = "segment-store-name-2";
 
     private final Statistics statisticsToAdd1 = mock(Statistics.class);
     private final Statistics statisticsToAdd2 = mock(Statistics.class);
-    private final Sensor sensor = createMock(Sensor.class);
-    private final StreamsMetricsImpl streamsMetrics = mock(StreamsMetricsImpl.class);
-    private final TaskId taskId = new TaskId(0, 0);
+
+    private final Sensor bytesWrittenToDatabaseSensor = createMock(Sensor.class);
+    private final Sensor bytesReadFromDatabaseSensor = createMock(Sensor.class);
+    private final Sensor memtableBytesFlushedSensor = createMock(Sensor.class);
+    private final Sensor memtableHitRatioSensor = createMock(Sensor.class);
+    private final Sensor writeStallDurationSensor = createMock(Sensor.class);
+    private final Sensor blockCacheDataHitRatioSensor = createMock(Sensor.class);
+    private final Sensor blockCacheIndexHitRatioSensor = createMock(Sensor.class);
+    private final Sensor blockCacheFilterHitRatioSensor = createMock(Sensor.class);
+    private final Sensor bytesReadDuringCompactionSensor = createMock(Sensor.class);
+    private final Sensor bytesWrittenDuringCompactionSensor = createMock(Sensor.class);
+    private final Sensor numberOfOpenFilesSensor = createMock(Sensor.class);
+    private final Sensor numberOfFileErrorsSensor = createMock(Sensor.class);
+
+    private final StreamsMetricsImpl streamsMetrics = niceMock(StreamsMetricsImpl.class);
+    private final RocksDBMetricsRecordingTrigger recordingTrigger = mock(RocksDBMetricsRecordingTrigger.class);
+    private final TaskId taskId1 = new TaskId(0, 0);
+    private final TaskId taskId2 = new TaskId(0, 2);
 
     private final RocksDBMetricsRecorder recorder = new RocksDBMetricsRecorder(METRICS_SCOPE, STORE_NAME);
 
+    @Before
+    public void setUp() {
+        expect(streamsMetrics.rocksDBMetricsRecordingTrigger()).andStubReturn(recordingTrigger);
+        replay(streamsMetrics);
+    }
+
     @Test
-    public void shouldSetStatsLevelToExceptDetailedTimers() {
+    public void shouldSetStatsLevelToExceptDetailedTimersWhenStatisticsIsAdded() {
```

Comment:
Just wanted to say I really like the way this unit test is written! With the right usage of mocks we would avoid having any time-dependent flakiness.